server/src/main/java/org/elasticsearch/cluster/routing/RoutingNode.java

```diff
@@ -242,20 +231,7 @@ public int numberOfShardsWithState(ShardRoutingState... states) {
      * @return List of shards
      */
     public List<ShardRouting> shardsWithState(ShardRoutingState state) {
```

Comment:
AFAICT this method is only used in two places in prod code, both of which could reasonably accept a `Stream<ShardRouting>` instead and avoid the need to copy into a new list.

Reply:
This has plenty of unit test usages that needs to be updated as well. Do you mind if I do it in a followup pr (possibly with moving some of this methods to a test code)?

Reply:
sounds good